docs/gen_ref_pages.py

```diff
@@ -0,0 +1,35 @@
+"""Generate the code reference pages and navigation."""
+
+from pathlib import Path
+
+import mkdocs_gen_files
+
+nav = mkdocs_gen_files.Nav()
+
+for path in sorted(Path("src").rglob("*.py")):
+    module_path = path.relative_to("src").with_suffix("")
+    doc_path = path.relative_to("src").with_suffix(".md")
+    full_doc_path = Path("reference", doc_path)
+
+    parts = tuple(module_path.parts)
+    if not "mdp" in parts:
+        continue  # todo: add other modules here once docstrings added
```

Comment:
You added a lot of docstrings in this pr. Is the if statement still necessary?

Reply:
Unfortunately yes 😅